cppcache/src/SerializationRegistry.hpp

```diff
@@ -130,6 +138,11 @@ class TheTypeMap : private NonCopyable {
   inline void unbindPdxType(const std::string& objFullName);
 
   void rebindPdxType(std::string objFullName, TypeFactoryMethodPdx func);
+
+ private:
+  void bind(TypeFactoryMethod func, DSCode id) {
+    bind(func, static_cast<int32_t>(id));
```

Comment:
You should make sure that `DataSerializable` , `DataSerializablePrimitive` and `DataSerializableFixedId` don't use the same map or they can walk on each others IDs. Previously they did some bit shift magic but we should now have distinct maps for the different types.